Compilation fails on Github Action: CPack error : Problem running WiX.

**Describe the bug**
The latest compilation fails on Github Actions images with the following error:

Source code used: [tag 0.9.0](https://github.com/end2endzone/ShellAnything/tree/0.9.0) + fixes from #157.
Build run url: https://github.com/end2endzone/ShellAnything/actions/runs/10342831169

Build log output:
```
============================================================================
Creating ShellAnything install package
============================================================================
MSBuild version 17.10.4+10fbfbf2e for .NET Framework

  sa.shared.vcxproj -> D:\a\ShellAnything\ShellAnything\build\bin\Release\sa.shared.lib
  file_explorer_renew.vcxproj -> D:\a\ShellAnything\ShellAnything\build\bin\Release\file_explorer_renew.exe
  libexprtk.vcxproj -> D:\a\ShellAnything\ShellAnything\build\bin\Release\libexprtk.dll
  sa.core.vcxproj -> D:\a\ShellAnything\ShellAnything\build\bin\Release\sa.core.dll
  Copying default configurations files database.	Copying register.bat script.	Copying unregister.bat script.	Copying magic.mgc.	Copying file.exe.	Copying zlib.dll.	Copying glog.
  sa.api.vcxproj -> D:\a\ShellAnything\ShellAnything\build\bin\Release\sa.api.dll
  sa.logger.glog.vcxproj -> D:\a\ShellAnything\ShellAnything\build\bin\Release\sa.logger.glog.dll
  sa.windows.vcxproj -> D:\a\ShellAnything\ShellAnything\build\bin\Release\sa.windows.dll
  sa.shellextension.vcxproj -> D:\a\ShellAnything\ShellAnything\build\bin\Release\sa.shellextension.dll
  sa.tests.vcxproj -> D:\a\ShellAnything\ShellAnything\build\bin\Release\sa.tests.exe
  Copying test configuration files database.
  sa_plugin_process.vcxproj -> D:\a\ShellAnything\ShellAnything\build\bin\Release\sa_plugin_process.dll
  sa_plugin_services.vcxproj -> D:\a\ShellAnything\ShellAnything\build\bin\Release\sa_plugin_services.dll
  sa_plugin_strings.vcxproj -> D:\a\ShellAnything\ShellAnything\build\bin\Release\sa_plugin_strings.dll
  sa_plugin_test_data.vcxproj -> D:\a\ShellAnything\ShellAnything\build\bin\Release\sa_plugin_test_data.dll
  sa_plugin_time.vcxproj -> D:\a\ShellAnything\ShellAnything\build\bin\Release\sa_plugin_time.dll
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(2165,5): warning : The referenced project 'Solution Items (dummy)' does not exist. [D:\a\ShellAnything\ShellAnything\build\ALL_BUILD.vcxproj]
  1>
  CPack: Create package using ZIP
  CPack: Install projects
  CPack: - Install project: ShellAnything [Release]
  CPack: Create package
  CPack: - package: D:/a/ShellAnything/ShellAnything/build/ShellAnything-0.9.0-win64.zip generated.
  CPack: Create package using WIX
  CPack: Install projects
  CPack: - Install project: ShellAnything [Release]
  CPack: Create package
EXEC : CPack error : Problem running WiX. Please check 'D:/a/ShellAnything/ShellAnything/build/_CPack_Packages/win64/WIX/wix.log' for errors. [D:\a\ShellAnything\ShellAnything\build\PACKAGE.vcxproj]
EXEC : CPack error : Fatal WiX Generator Error [D:\a\ShellAnything\ShellAnything\build\PACKAGE.vcxproj]
EXEC : CPack error : Problem compressing the directory [D:\a\ShellAnything\ShellAnything\build\PACKAGE.vcxproj]
EXEC : CPack error : Error when generating package: ShellAnything [D:\a\ShellAnything\ShellAnything\build\PACKAGE.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(155,5): error MSB3073: The command "setlocal [D:\a\ShellAnything\ShellAnything\build\PACKAGE.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(155,5): error MSB3073: cd D:\a\ShellAnything\ShellAnything\build [D:\a\ShellAnything\ShellAnything\build\PACKAGE.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(155,5): error MSB3073: if %errorlevel% neq 0 goto :cmEnd [D:\a\ShellAnything\ShellAnything\build\PACKAGE.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(155,5): error MSB3073: D: [D:\a\ShellAnything\ShellAnything\build\PACKAGE.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(155,5): error MSB3073: if %errorlevel% neq 0 goto :cmEnd [D:\a\ShellAnything\ShellAnything\build\PACKAGE.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(155,5): error MSB3073: "C:\Program Files\CMake\bin\cpack.exe" -C Release --config ./CPackConfig.cmake [D:\a\ShellAnything\ShellAnything\build\PACKAGE.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(155,5): error MSB3073: if %errorlevel% neq 0 goto :cmEnd [D:\a\ShellAnything\ShellAnything\build\PACKAGE.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(155,5): error MSB3073: :cmEnd [D:\a\ShellAnything\ShellAnything\build\PACKAGE.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(155,5): error MSB3073: endlocal & call :cmErrorLevel %errorlevel% & goto :cmDone [D:\a\ShellAnything\ShellAnything\build\PACKAGE.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(155,5): error MSB3073: :cmErrorLevel [D:\a\ShellAnything\ShellAnything\build\PACKAGE.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(155,5): error MSB3073: exit /b %1 [D:\a\ShellAnything\ShellAnything\build\PACKAGE.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(155,5): error MSB3073: :cmDone [D:\a\ShellAnything\ShellAnything\build\PACKAGE.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(155,5): error MSB3073: if %errorlevel% neq 0 goto :VCEnd [D:\a\ShellAnything\ShellAnything\build\PACKAGE.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(155,5): error MSB3073: :VCEnd" exited with code 1. [D:\a\ShellAnything\ShellAnything\build\PACKAGE.vcxproj]
```


**To Reproduce**
N/A

**Expected behavior**
Compilation succeeds

**Screenshots**
N/A

**Environment**
Runner config:
```
Current runner version: '2.317.0'
Operating System
  Microsoft Windows Server [2](https://github.com/end2endzone/ShellAnything/actions/runs/10342428095/job/28625368972#step:1:2)022
  10.0.20348
  Datacenter
Runner Image
  Image: windows-2022
  Version: 20240807.2.0
  Included Software: https://github.com/actions/runner-images/blob/win22/20240807.2/images/windows/Windows2022-Readme.md
  Image Release: https://github.com/actions/runner-images/releases/tag/win22%2F20240807.2
Runner Image Provisioner
  2.0.[3](https://github.com/end2endzone/ShellAnything/actions/runs/10342428095/job/28625368972#step:1:3)74.1
```

**Additional context**
N/A

--- a/.github/workflows/build_windows.yml
+++ b/.github/workflows/build_windows.yml
@@ -89,6 +89,11 @@ jobs:
         gist_file_name: ${{env.GIT_REPOSITORY}}.${{env.GIT_BRANCH}}.${{env.GITHUB_WORKFLOW}}.json
         file_path: ./ci/github/tests_not_available.badge.json
 
+    - name: Install CMake (issue \#158)
+      working-directory: ${{env.GITHUB_WORKSPACE}}
+      shell: cmd
+      run: call ci\github\install_cmake.bat
+
     - name: Install Doxygen
       working-directory: ${{env.GITHUB_WORKSPACE}}
       shell: cmd

--- a/CHANGES
+++ b/CHANGES
@@ -1,3 +1,9 @@
+Changes for 0.10.0
+
+* Fixed issue #157: Compilation fails on Github Action: `fatal error C1083: Cannot open include file: 'atlbase.h': No such file or directory`.
+* Fixed issue #158: Compilation fails on Github Action: `CPack error : Problem running WiX.`.
+
+
 Changes for 0.9.0
 
 * Fixed issue #111: Move code from the shell extension to the core.

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -114,7 +114,7 @@ The following software must be installed on the system for compiling source code
 * [exprtk d312ba9](https://github.com/ArashPartow/exprtk) (downloaded automatically)
 * [File and libmagic for Windows v5.38 + modifications](https://github.com/Cirn09/file-windows/tree/39f1624b4c95a0ab657a1084d50069270ebb8947)
 * [zlib v1.2.11](https://github.com/madler/zlib/releases/tag/v1.2.11)
-* [CMake v3.4.3](http://www.cmake.org/) (or newer)
+* [CMake v3.4.3](http://www.cmake.org/) (tested up to 3.29.7)
 * [imneme/pcg-cpp](https://github.com/imneme/pcg-cpp/tree/428802d1a5634f96bcd0705fab379ff0113bcf13) (or newer)
 * (optional) [Grip (GitHub Readme Instant Preview)](https://github.com/joeyespo/grip)  v4.5.2 (or newer)
 
@@ -122,7 +122,7 @@ The following software must be installed on the system for compiling source code
 ### Windows Requirements ###
 
 * Microsoft Visual C++ 2017 or newer
-* (optional) [WiX Toolset](https://wixtoolset.org/)  v3.11.2 (or newer)
+* (optional) [WiX Toolset](https://wixtoolset.org/)  v3.11.2 (tested up to 3.14.1.8722)
 
 
 

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -62,6 +62,7 @@ matrix:
 before_build:
 - cmd: call %APPVEYOR_BUILD_FOLDER%\ci\appveyor\list_services_status.bat
 - cmd: call %APPVEYOR_BUILD_FOLDER%\ci\appveyor\list_registry_keys.bat
+- cmd: call %APPVEYOR_BUILD_FOLDER%\ci\appveyor\install_cmake.bat
 - cmd: call %APPVEYOR_BUILD_FOLDER%\ci\appveyor\install_doxygen.bat
 - cmd: call %APPVEYOR_BUILD_FOLDER%\ci\appveyor\install_googletest.bat
 - cmd: call %APPVEYOR_BUILD_FOLDER%\ci\appveyor\install_tinyxml2.bat

--- a/ci/appveyor/install_cmake.bat
+++ b/ci/appveyor/install_cmake.bat
@@ -1,0 +1,11 @@
+@echo off
+
+:: Validate appveyor's environment
+if "%APPVEYOR_BUILD_FOLDER%"=="" (
+  echo Please define 'APPVEYOR_BUILD_FOLDER' environment variable.
+  exit /B 1
+)
+
+:: Call matching script for windows
+call "%APPVEYOR_BUILD_FOLDER%\ci\windows\%~n0.bat"
+if %errorlevel% neq 0 exit /b %errorlevel%

--- a/ci/github/install_cmake.bat
+++ b/ci/github/install_cmake.bat
@@ -1,0 +1,11 @@
+@echo off
+
+:: Validate GitHub CI's environment
+if "%GITHUB_WORKSPACE%"=="" (
+  echo Please define 'GITHUB_WORKSPACE' environment variable.
+  exit /B 1
+)
+
+:: Call matching script for windows
+call "%GITHUB_WORKSPACE%\ci\windows\%~n0.bat"
+if %errorlevel% neq 0 exit /b %errorlevel%

--- a/ci/windows/install_cmake.bat
+++ b/ci/windows/install_cmake.bat
@@ -1,0 +1,78 @@
+@echo off
+
+:: Install CMake.
+:: https://cmake.org/download/
+:: https://github.com/Kitware/CMake/releases/latest
+
+echo Installing CMake...
+set CMAKE_MSI_URL=https://github.com/Kitware/CMake/releases/download/v3.29.7/cmake-3.29.7-windows-x86_64.msi
+set CMAKE_MSI_FILE=%TEMP%\cmake-3.29.7-windows-x86_64.msi
+set CMAKE_MSI_INSTALL_DIR=C:\Program Files\CMake
+
+:: Validate if python launcher is installed
+where py.exe >NUL 2>NUL
+if errorlevel 1 (
+  echo Command failed. Please install python and python launcher to continue.
+  exit /B %errorlevel%
+)
+echo Found python interpreter
+python --version
+
+:: Validate if pip is installed
+where pip.exe >NUL 2>NUL
+if errorlevel 1 (
+  echo Command failed. Please install pip ^(Package Installer Python^) to continue.
+  exit /B %errorlevel%
+)
+
+:: Download installer
+echo Downloading %CMAKE_MSI_URL%...
+python -c "import urllib.request; import os; urllib.request.urlretrieve(os.getenv('CMAKE_MSI_URL'),os.getenv('CMAKE_MSI_FILE'));"
+if errorlevel 1 (
+  echo Command failed. Unable to download cmake installer.
+  exit /B %errorlevel%
+)
+if NOT EXIST "%CMAKE_MSI_FILE%" (
+  echo Download failed. The file '%CMAKE_MSI_FILE%' is not found.
+  exit /B %errorlevel%
+)
+echo done.
+echo.
+
+:: Validate administrative permissions...
+:: https://stackoverflow.com/questions/4051883/batch-script-how-to-check-for-admin-rights
+echo Checking administrative permissions...
+net session >nul 2>&1
+if %errorLevel% == 0 (
+  echo Found administrative permissions. Installation in quiet mode enabled.
+  set QUIET_MODE_ARGS=/quiet /qn
+) else (
+  echo Administrative permissions not found. Installation in interractive mode will be used.
+)
+
+:: Installing MSI
+REM Note: `/quiet /qn` is not available when launched in user mode.
+echo Running CMake installer...
+echo QUIET_MODE_ARGS=%QUIET_MODE_ARGS%
+msiexec /i "%CMAKE_MSI_FILE%" %QUIET_MODE_ARGS% /norestart /log "%TEMP%\cmake_insall.log"
+if errorlevel 1 (
+  echo Failed to install MSI package "%CMAKE_MSI_FILE%".
+  exit /B %errorlevel%
+)
+echo done.
+echo.
+
+:: Searching for cmake executable
+echo Searching for cmake executable...
+set PATH=%CMAKE_MSI_INSTALL_DIR%\bin;%PATH%
+where cmake.exe 1>NUL 2>NUL
+if errorlevel 1 (
+  echo Error: CMake not installed on system.
+  exit /B %errorlevel%
+)
+echo CMake found.
+cmake --version
+echo.
+
+echo CMake was installed on the system without error.
+echo.

